datastore/load.go

```diff
@@ -302,11 +302,15 @@ func setVal(v reflect.Value, p Property) (s string) {
 			return fmt.Sprintf("%v is unsettable", v.Type())
 		}
 
-		rpValue := reflect.ValueOf(pValue)
-		if !rpValue.Type().AssignableTo(v.Type()) {
-			return fmt.Sprintf("%q is not assignable to %q", rpValue.Type(), v.Type())
+		if pValue == nil {
+			v.Set(reflect.Zero(v.Type()))
```

Comment:
it might be nice to add a quick comment here explaining why we are converting nil to Zero

Reply:
Added